### PR TITLE
Added Gallery, Thumbnail components

### PR DESCRIPTION
**Pull request by tjunxiang92**
Added the following
- Gallery Component
- Thumbnail Component
- Delete Button fixes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,7 +60,46 @@ Simply `npm install vue-fineuploader@next` and see the documentation for your sp
...
 
-Not implemented yet.
+```javascript
+<template>
+  <Gallery :uploader="uploader" />
+</template>
+
+<script>
+  import FineUploaderTraditional from 'fine-uploader-wrappers';
+  import Gallery from '../modules/vue-fineuploader/src/gallery';
```
Comment (Elhebert) [inline README.md]: Could you remove the `;` here ? For consistency.

Also, `Gallery` should be loaded from 'vue-fineuploader/gallery'. (I have to update the README.md for all the other components)

File: README.md [hunk 2]
```diff
@@ -60,7 +60,46 @@ Simply `npm install vue-fineuploader@next` and see the documentation for your sp
...
+      const uploader = new FineUploaderTraditional({
+        options: {
+          deleteFile: {
+            enabled: true,
+            endpoint: 'http://localhost:8000/uploads'
+          },
+          request: {
+              endpoint: 'http://localhost:8000/uploads',
+              // Used for AWS
+              // accessKey: 'JB/Ckd2bcqydVOYhbbAZo+kTgGWzeXQt9UacmaC7',
```
Comment (Elhebert) [inline README.md]: I guess this shouldn't be there.

And, for the endpoint, you can use `my/upload/endpoint`, for consistency with the rest of the README

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "vue-fineuploader",
-  "version": "2.0.0-rc.6",
+  "version": "2.1.0-rc.6",
```
Comment (Elhebert) [inline package.json]: This was supposed to be a release candidate until all the components are finished, and be officially pushed in 2.0.0

Thus, incremented the `rc`, not the minor version.

File: src/gallery/FileInputComponent.vue [hunk 4]
```diff
@@ -0,0 +1,36 @@
+<template>
+  <FileInput 
+    class='react-fine-uploader-gallery-file-input-container'
```
Comment (Elhebert) [inline src/gallery/FileInputComponent.vue]: Avoid using `react` in the class name. Use `vue` instead.

File: src/gallery/index.vue [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+    <FileInputComponent 
+      :multiple="fileInputMultiple"
+      :uploader="uploader"
+    />
+    <ProgressBar 
+      class='react-fine-uploader-gallery-total-progress-bar'
+      :uploader="uploader"
+    />
+
+    <!-- <ReactCssTransitionGroup
```
Comment (Elhebert) [inline src/gallery/index.vue]: This shouldn't be here.

**Comment by tjunxiang92** [comment]
Appreciate the time taken out to test and comment on the code :) Made the necessary changes.

File: README.md [hunk 6]
```diff
@@ -60,10 +60,147 @@ Simply `npm install vue-fineuploader@next` and see the documentation for your sp
...
+```
+
+##### Todos
+- Implement animation
+- Fix progress bar
+- Fix slot for content
 
 ### Low-level Components
 
+#### Script
```
Comment (Elhebert) [inline README.md]: I assume this was misplaced ?

File: README.md [hunk 7]
```diff
@@ -723,5 +860,73 @@ Note: This assumes you have additional components or code to allow files to actu
...
+Suppose you wanted to render a thumbnail for each file as new files are submitted to Fine Uploader. Your Vue component may look like this:
+
+Note: This assumes you have additional components or code to allow files to actually be submitted to Fine Uploader.
+
+```js
+<template>
+  <div v-for="file in state.submittedFiles">
+    <thumbnail :id="file" :uploader="uploader" />
+    <cancel-button :id="file" :uploader="uploader" />
+    <delete-button :id="file" :uploader="uploader" />
```
Comment (Elhebert) [inline README.md]: I would remove the `delete` and the `cancel-button`, they're not needed in the `thumbnail` part of the README.

**Comment by Elhebert** [comment]
I've got a big work-related project release in 2 weeks, so I'll try to find some time this week-end to test your PR and finish the review.

But you did a really great job here ! 

Thanks @tjunxiang92 !

File: README.md (hunk 7 above)
Comment (tjunxiang92) [reply README.md]: Great catch! Thanks

**Comment by tjunxiang92** [comment]
@Elhebert Thanks for taking time out to check it. New to contributing in open source and open to feedback on places I can improve in :)

All the best for your big work-related project release in 2 weeks!

**Comment by Elhebert** [comment]
@tjunxiang92 Just got some time to test this, and it's working pretty well. Nice job!

I'm merging this.

**Comment by tjunxiang92** [comment]
Sweet! Updated onto develop branch.

**Comment by Elhebert** [comment]
I have some update to do in the documentation, then I'm merging all of it onto master, and publish it on npm ;)

**Pull request merged**